### ISSUE TITLE
Support for Windows ssh-agent

### ISSUE THREAD

**Issue by sophaskins** [facebook/nuclide#100]
Hiya!

I set up Nuclide on a Windows machine (Windows 8.1, Github for Windows now installed), and am trying to use nuclide-remote-projects to connect to a remote server that I normally develop on via SSH.

I have an SSH key with a passphrase, and normally I use Pageant from the PuTTY tools for SSH on Windows (as far as I know, far-and-away the most common way to SSH from Windows). While I have that agent running, Atom does not seem to be connecting to it (perhaps it's relying on the SSH_AUTH_SOCK env variables that ssh-agent sets?)

I'd like to be able to use this (common) key agent on Windows. Some possibilities for that are:

* connecting with Pageant's protocol (example: https://github.com/paramiko/paramiko/blob/master/paramiko/win_pageant.py). This would involve writing some Windows code, but would be quite slick
* some instructions on setting up ssh-pageant or similar (https://github.com/cuviper/ssh-pageant) and how to get Atom to launch with env variables pointing to the right thing.

Thanks!

**Comment by Shawty81** [comment]
When trying to do the same (trying to use Pageant) I get a message in Atom saying: "All configured authentication methods failed"

My colleague checked the server logs for me and found the following message: "Received disconnect from %my IP%: 11: [preauth]"

Not sure what's going on here, but we'll give it another look tomorrow...

**Comment by bolinfest** [comment]
Hi there, if you're comfortable tweaking the code, can you let me know if the following fixes things for you:

Go to https://github.com/facebook/nuclide/blob/48b6dd525fdad10a63f412e42521bd163b9563a8/pkg/nuclide/remote-connection/lib/SshHandshake.js#L110

and modify it from:

```js
agent: process.env.SSH_AUTH_SOCK,
```

to:

```
agent: process.env['SSH_AUTH_SOCK'] || 'pageant',
```

If that works for you, then I'll commit that to the repo.

**Comment by sophaskins** [comment]
That does in fact fix my SSH issues! Thank you very much, @bolinfest!

**Comment by bolinfest** [comment]
No problem -- thanks for testing! Proper fix in 7262b88599ba4dbf7396d33b7d2718a048c9f8d9.
